src/aap_eda/api/views/eda_credential.py

```diff
@@ -220,15 +220,11 @@ def destroy(self, request, *args, **kwargs):
                 {"errors": error}, status=status.HTTP_400_BAD_REQUEST
             )
 
-        # If the credential is in use and the 'force' flag
-        # is not True, raise a PermissionDenied exception
-        is_used = models.Activation.objects.filter(
-            decision_environment__eda_credential=eda_credential
-        ).exists()
+        references = get_references(eda_credential)
 
-        if is_used and not force:
+        if bool(references) and not force:
             raise exceptions.Conflict(
-                "Credential is being used by Activations "
+                "Credential is being referenced by resources "
```

Comment:
@hsong-rh I think this would be a better message.
```
Credential {name} is being referenced by other resources and cannot be deleted. If you want to force delete, please add the /?force=true query param.
```